extensions/amp-gist/0.1/amp-gist.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Embeds a Github gist
+ *
+ * Example:
+ * <code>
+ * <amp-gist
+ *   layout="responsive"
+ *   data-gistid="a19e811dcd7df10c4da0931641538497"
+ *   width="100"
+ *   height="100">
+ * </amp-gist>
+ * </code>
+ */
+
+import {getIframe} from '../../../src/3p-frame';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {removeElement} from '../../../src/dom';
+import {user} from '../../../src/log';
+
+export class AmpGist extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?HTMLIFrameElement} */
+    this.iframe_ = null;
+  }
+
+  /**
+  * @param {boolean=} opt_onLayout
+  * @override
+  */
+  preconnectCallback(opt_onLayout) {
+    this.preconnect.url('https://gist.github.com/', opt_onLayout);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /**@override*/
+  layoutCallback() {
+    const gistid = user().assert(
+      this.element.getAttribute('data-gistid'),
+      'The data-gistid attribute is required for <amp-gist> %s',
+      this.element);
+    const width = this.element.getAttribute('width');
+    const height = this.element.getAttribute('height');
+    const url = 'https://gist.github.com/';
+
+    const iframe = getIframe(this.win, this.element, 'gist');
+
+    iframe.setAttribute('frameborder', 'no');
+    iframe.setAttribute('scrolling', 'no');
+
+    const src = url + encodeURIComponent(gistid) + '.pibb';
+
+    iframe.src = src;
+
+    this.applyFillContent(iframe);
+    iframe.width = width;
```

Comment:
Unfortunately we can't do this. We need to resize the iframe after the fact to become the same height as the actual gist. This part is a bit tricky, we need to take a similar approach as `amp-facebook` (meaning we can't use `.pibb` either)

- [ ] Need to have a `github.js` file in `3p` folder. That file runs **_inside_** the 3P iframe. Inside that file, we would load `https://gist.github.com/<gist>.js`. Note the `.js` instead of `.pibb`.
- [ ] When `https://gist.github.com/<gist>.js` loads, it automatically does a  `document.write` of the gist content. We need to measure the height of the document after this script does its thing and then issue a `context.updateDimensions(newHeight, newWidth)` from the `3p/github.js`
- [ ] inside `amp-gist`, we need to listen to the resize event and update the dimension of the iframe:
```
    // Triggered by context.updateDimensions() inside the iframe.
    listenFor(iframe, 'embed-size', data => {
      this./*OK*/changeHeight(data.height);
    }, /* opt_is3P */true);
```

@levidurfee This is a bit of work, totally up to you if you like to do this part or not. If we don't do this part immediately, we can restrict the layout to just `fixed` (just update `isLayoutSupported`) for now and merge the PR, then create an issue to fix this before the component goes out of experimental state.